### DISCUSSION TITLE
Error 502 with Supabase client when using Nginx as a reverse proxy

### DISCUSSION THREAD

**Discussion by sirtrade** [supabase/supabase#29211] [Feature Requests]
Description:
I encountered an issue with Supabase in my Next.js application when using Nginx as a reverse proxy. The application was returning a 502 error when attempting to set cookies with the Supabase client in the middleware. After investigating, it turned out that Nginx had limitations on header size and request timeouts, which caused the issue.

Steps to reproduce:
1. Set up a Next.js app with Supabase client in the middleware.
2. Use Nginx as a reverse proxy for the app.
3. Set cookies using Supabase's `setAll` method in the middleware.
4. Observe the 502 error.

Resolution:
The issue was resolved by increasing the `large_client_header_buffers` and timeout settings in Nginx config: 
```
    large_client_header_buffers 4 16k;
    proxy_connect_timeout 60s;
    proxy_send_timeout 60s;
    proxy_read_timeout 60s;
    send_timeout 60s;
    proxy_buffer_size   128k;
    proxy_buffers       4 256k;
    proxy_busy_buffers_size 256k;
    client_max_body_size 10M;
```

Would it be possible to include a note in the documentation about this issue, or improve how Supabase interacts with proxy servers like Nginx in terms of header size and timeouts?

I hope this information helps someone.

**Comment by R1NEJ0** [comment]
I've been fighting with this error.

Thanks!!!!

**Comment by 539hex** [comment]
Thanks so much!

**Comment by YouMingYeh** [comment]
Bro you saved my life

**Comment by BonLevain** [comment]
@fullgream This is super helpful.  Do you mind elaborating on _why_ this fixes the issue?  Just trying to understand how to figured out this was needed, and wanting to learn a bit more about supabase/nginx in the process.

**Reply by sirtrade** [reply]
To be honest, I haven't really looked into what exactly is going on inside. I guess it's all about nginx's default value restrictions

**Comment by cbuchele** [comment]
Thanks so much!!! Saved me tons of work!

**Reply by olelundmaegbaek** [reply]
Thank you!

**Comment by fetouh-sadegh** [comment]
DUDE, thank you!
I spent 4 hours trying everything, asking GPT, Stack Overflow, etc., and the project deadline is tomorrow lol. So, thanks!

**Comment by KubaZachacz** [comment]
OMG, a hero! 
Yet another reason why Human Intelligence wins with an AI, 
I could figure it out with agent's for an hours... 
and then... just first google result fixes it in a minute ❤️
